cmd/promoted-image-governor/main.go

```diff
@@ -270,6 +275,53 @@ func isMirroredFromOCP(tag api.ImageStreamTagReference, refs []ImageStreamRef) b
 	return false
 }
 
+type refGetter interface {
+	GetRef(org, repo, ref string) (string, error)
+}
+
+type gitRefGetter struct {
+}
+
+func (g *gitRefGetter) GetRef(org, repo, ref string) (ret string, retError error) {
+	return util.GetRemoteBranchCommitSha(org, repo, ref)
+}
+
+// checkImageStreamTags checks if promotedTags are built with the current head of org/repo/branch
+func checkImageStreamTags(ctx context.Context, client ctrlruntimeclient.Client, promotedTags map[api.ImageStreamTagReference]api.Metadata, refGetter refGetter) []error {
+	var errs []error
+	for promotedTag, metadata := range promotedTags {
+		ist := &imagev1.ImageStreamTag{}
+		istName := fmt.Sprintf("%s:%s", promotedTag.Name, promotedTag.Tag)
+		if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: promotedTag.Namespace, Name: istName}, ist); err != nil {
```

Comment:
Not sure if its a great idea to put this into a CLI, this will do # of promoted imagestreamtag api calls, which I believe is a high 4 digit/low 5 digit number?

It might be simpler to add this as a metric to the controller, it has the data cached anyways.